server/src/printers/KleeConstraintsPrinter.cpp

```diff
@@ -148,37 +111,42 @@ void KleeConstraintsPrinter::genConstraintsForPointerInStruct(const ConstraintsS
 
 void KleeConstraintsPrinter::genConstraintsForStruct(const ConstraintsState &state) {
     StructInfo curStruct = typesHandler->getStructInfo(state.curType);
-
+    bool isStruct = curStruct.subType == SubType::Struct;
     for (const auto &field : curStruct.fields) {
         std::string errorMessage = "Unrecognized field of type '" + field.type.typeName() +
                                    "' in struct '" + curStruct.name + "'.";
         auto access = PrinterUtils::getFieldAccess(state.curElement, field);
-        ConstraintsState newState = { state.paramName, access, field.type, state.endString, state.depth + 1 };
+        ConstraintsState newState = { state.paramName,
+                                      access,
+                                      field.type,
+                                      isStruct ? state.endString : false,
```

Comment:
That is still essential difference between struct and union. Need to be investigated (in progress).